### ISSUE TITLE
GameLauncher fails to load levels when built in Pre-built SDK engine configuration

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#15166]
**Describe the bug**
Attempting to load a level in the project's GameLauncher built in Pre-built SDK engine configuration results in the following error being printed in the console:
```
(System) - O3DE could not initialize correctly for the following reason(s):
Element 'NULL'(0x41405e39) with class ID '{E2DC7EB8-02D1-4E6D-BFE4-CE652FCB7C7F}' found in 'AZStd::vector<AZ::Component*, allocator>' is not registered with the serializer!
If this class was removed, consider using serializeContext->ClassDeprecate(...) to register classes as having been deprecated to avoid this error.  File levels/test2/test2.spawnable
=== Serialize stack ===
[ Class: 'AzFramework::Spawnable' Version: 2 Address: 000001DF458F0DA0 Uuid: {855E3021-D305-4845-B284-20C3F7FDF16B} ]
[ Class: 'AZStd::vector<AZStd::unique_ptr<AZ::Entity>, allocator>' Version: 0 Address: 000001DF458F0E50 Uuid: {C91F807C-62E6-5D70-A10F-4C6BB05C0321} ]
[ Class: 'AZStd::unique_ptr<AZ::Entity>' Version: 0 Address: 000001DF50B590F8 Uuid: {8FFB6D85-994F-5262-BA1C-D0082A7F65C5} ]
[ Class: 'AZ::Entity' Version: 2 Address: 000001DF3D560538 Uuid: {75651658-8663-478D-9090-2432DFCAFA44} ]
[ Class: 'AZStd::vector<AZ::Component*, allocator>' Version: 0 Address: 000001DF3D560548 Uuid: {13D58FF9-1088-5C69-9A1F-C2A144B57B78} ]
```

The Camera entity is not working correctly and no models present in the level can be seen. It has been verified using the default level as well as using modified levels.
Attempting to load level using _autoexec.game.setreg_ file results in the same issue with the error being displayed in a prompt window.

This issue occurs on both Windows and Linux, as well as both Stabilization and Development branches.

Please refer to the attached screenshots and log for more information.

**Prerequisites**

- O3DE built in Pre-built SDK engine.
- A new project using default template is built.

**Steps to reproduce**
1. Launch the Editor.
2. Create a new level and save it.
3. Close the Editor.
4. Navigate to _\Users\<user_name>\O3DE\Projects\<project_name>\build\windows\bin\profile_ and launch the <project_name>.GameLauncher.

**Expected behavior**
GameLauncher loads levels correctly making them functional.

**Actual behavior**
GameLauncher fails to load levels making them nonfunctional.

**Screenshots/Video**
![GameLauncherLoadLevel](https://user-images.githubusercontent.com/86953659/225900186-7aecb17f-0d9f-4213-bb52-aa0571f73b03.PNG)
![GameLauncherAutoexec](https://user-images.githubusercontent.com/86953659/225900206-8fcb8071-b4b9-4318-85b1-ca39e40613e9.PNG)

**Found in Branch**
Stabilization/2305

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
f1fbf4b

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10
- CPU: AMD Ryzen 5 3600
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[Game.log](https://github.com/o3de/o3de/files/11001108/Game.log)

**Comment by LB-KacperKapusta** [comment]
@lmbr-pip, any level that I've created and attempted to load has failed in the described configuration. The same issue has occurred when loading _prefabs/default_level.spawnable_ using the ImGui.

**Comment by LB-MichalRodzos** [comment]
We noticed that the same error is printed in the console when MultiplayerSample.ServerLauncher (from MultiplayerSample Project) has been run without any arguments via CLI and the `LoadLevel Levels/NewStarbase/NewStarbase.spawnable` command was used to load level. It was observed in Stabilization/2305 Installer build on Windows and Linux.

**Comment by lumberyard-employee-dm** [comment]
@AMZN-alexpete Ok, So the issues is that the list of required gems set by the engine via the "gem_names" field isn't being replicated in the generated `engine.json` that is part of the SDK layout.
![image](https://user-images.githubusercontent.com/56135373/232921970-8dbafc0b-4593-4717-ba5d-6f617f138c39.png)

We need to update the engine.json configure template for the SDK layout [engine.json.in](https://github.com/o3de/o3de/blob/development/cmake/install/engine.json.in#L1-L13) file to add the engine gems to the generated file.

That would should solve the issue and have the Camera gem and the other required active gems be loaded.

**Comment by AMZN-alexpete** [comment]
Good catch!

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2305 (c351295) build.
